test/gtest/lstm_extra.cpp

```diff
@@ -0,0 +1,156 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "lstm.hpp"
+#include "get_handle.hpp"
+#include <miopen/env.hpp>
+#include <gtest/gtest.h>
+#include <boost/algorithm/string.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+static bool SkipTest(void) { return miopen::IsDisabled(ENV(MIOPEN_TEST_ALL)); }
+
+void GetArgs(const std::string& param, std::vector<std::string>& tokens)
+{
+    std::stringstream ss(param);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class ConfigWithFloat : public testing::TestWithParam<std::vector<std::string>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<std::string> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf:
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenHalf, miopenInt8, miopenBFloat16, miopenInt32, miopenDouble "
+                  "data type not supported by "
+                  "lstm_extra test";
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+
+        testing::internal::CaptureStderr();
+        test_drive<lstm_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+bool IsTestSupportedForDevice(const miopen::Handle& handle)
+{
+    std::string devName = handle.GetDeviceName();
+    if(devName == "gfx900" || devName == "gfx906" || devName == "gfx908" || devName == "gfx90a" ||
+       miopen::StartsWith(devName, "gfx94") || miopen::StartsWith(devName, "gfx103") ||
+       miopen::StartsWith(devName, "gfx110"))
+        return true;
+    else
+        return false;
+}
+
+TEST_P(ConfigWithFloat, FloatTest)
+{
+    const auto& handle = get_handle();
+    if(IsTestSupportedForDevice(handle) && !SkipTest())
+    {
+        Run2dDriver(miopenFloat);
+    }
+    else
+    {
+        GTEST_SKIP();
+    }
+};
+
+std::vector<std::string> GetTestCases(void)
+{
+    std::string flags       = "test_lstm --verbose ";
+    std::string commonFlags = "--batch-size 32 --seq-len 3 --batch-seq 32 32 32 --vector-len 128 "
+                              "--hidden-size 128 --num-layers 1 --in-mode 0 --bias-mode 0";
+
+    const std::vector<std::string> test_cases = {
+        // clang-format off
+    {flags + commonFlags + " -dir-mode 0 --no-hx"},
+    {flags + commonFlags + " -dir-mode 0 --no-dhy"},
```

Comment:
Let's avoid disabling clang-format, use EOL comments to control formatting
```suggestion
    const std::vector<std::string> test_cases = { //
        {flags + commonFlags + " -dir-mode 0 --no-hx"}, //
        {flags + commonFlags + " -dir-mode 0 --no-dhy"}, //
```
etc

Reply:
enabled clang format

Reply:
[resolved]